apps/vscode/vscode.py

```diff
@@ -170,53 +170,183 @@ def command_palette():
+        _replace_tab_with_untitled_file()
 
-    def split_reset():
-        actions.user.vscode("workbench.action.evenEditorWidths")
+    def split_create_down():
+        actions.user.split_clone_down()
+        _replace_tab_with_untitled_file()
 
-    def split_last():
+    def split_create_up():
+        actions.user.split_clone_up()
+        _replace_tab_with_untitled_file()
+
+    ## Duplicated splits
+    def split_clone():
+        actions.user.vscode("workbench.action.splitEditor")
+
+    def split_clone_right():
+        actions.user.vscode("workbench.action.splitEditorRight")
+
+    def split_clone_left():
+        actions.user.vscode("workbench.action.splitEditorLeft")
+
+    def split_clone_down():
+        actions.user.vscode("workbench.action.splitEditorDown")
+
+    def split_clone_up():
+        actions.user.vscode("workbench.action.splitEditorUp")
+
+    def split_clone_vertically():
+        actions.user.vscode("workbench.action.splitEditor")
+
+    def split_clone_horizontally():
+        actions.user.vscode("workbench.action.splitEditorOrthogonal")
+
+    # Destruction
+    def split_close():
+        actions.user.vscode("workbench.action.joinTwoGroups")
+
+    def split_close_all():
+        actions.user.vscode("workbench.action.editorLayoutSingle")
+
+    # Navigation
+    def split_focus_previous():
         actions.user.vscode("workbench.action.focusLeftGroup")
 
-    def split_next():
+    def split_focus_next():
         actions.user.vscode_and_wait("workbench.action.focusRightGroup")
 
-    def split_window_down():
-        actions.user.vscode("workbench.action.moveEditorToBelowGroup")
+    def split_focus_first():
+        actions.user.vscode("workbench.action.focusFirstEditorGroup")
 
-    def split_window_horizontally():
-        actions.user.vscode("workbench.action.splitEditorOrthogonal")
+    def split_focus_final():
+        actions.user.vscode("workbench.action.focusLastEditorGroup")
 
-    def split_window_left():
+    # Arrangement
+    def split_move_down():
+        actions.user.vscode("workbench.action.moveEditorToBelowGroup")
+
+    def split_move_left():
         actions.user.vscode("workbench.action.moveEditorToLeftGroup")
 
-    def split_window_right():
+    def split_move_right():
         actions.user.vscode("workbench.action.moveEditorToRightGroup")
 
-    def split_window_up():
+    def split_move_up():
         actions.user.vscode("workbench.action.moveEditorToAboveGroup")
 
-    def split_window_vertically():
-        actions.user.vscode("workbench.action.splitEditor")
+    # Resizing
+    def split_toggle_orientation():
+        actions.user.vscode("workbench.action.toggleEditorGroupLayout")
 
-    def split_window():
-        actions.user.vscode("workbench.action.splitEditor")
+    def split_toggle_maximize():
+        actions.user.vscode("workbench.action.maximizeEditor")
+
+    def split_layout_reset():
+        actions.user.vscode("workbench.action.evenEditorWidths")
```

Comment:
workbench.action.editorLayoutSingle

Reply:
I don't agree with this one. As the action description says, it's meant to resize all splits to the default, which is what workbench.action.evenEditorWidths does. What you're suggesting would be suitable for split_close_all() (and is what it does).